 **a**. **Build 99% Confidence Interval Using Sample Standard Deviation**

Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.


A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:

1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29


In [1]:
# importing necessary libraries
import numpy as np
from scipy import stats
# given data
CI=0.99
alpha=0.01
n=15
df=n-1
# x_bar=np.mean(sample_data)

In [2]:
# stores the sample data in a array
sample_data=np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

In [3]:
x_bar=np.mean(sample_data) # calculating sample mean

In [4]:
std=np.std(sample_data,ddof=1) # calculates sample standard deviation

In [5]:
print(x_bar) # prints sample mean
print(std) # prints sample standard deviation

1.2386666666666666
0.19316412956959936


In [6]:
# Calculates the critical value of the t-distribution for a two-tailed test
t_critical=stats.t.ppf(1-alpha/2,df)


In [7]:
t_critical # prints t_critical value

2.976842734370834

In [8]:
Margin_of_Error=t_critical*(std/np.sqrt(n)) # calculates the margin of error

In [9]:
Margin_of_Error # prints margin of error


0.14846932822817596

In [10]:
lower_limit=x_bar-Margin_of_Error # calculates the lower limit
upper_limit=x_bar+Margin_of_Error # calculates the upper limit

In [11]:
print(lower_limit) # prints lower limit
print(upper_limit) # prints upper limit

1.0901973384384906
1.3871359948948425


In [12]:
# calculating confidence_interval
confidence_interval=(lower_limit,upper_limit)

In [13]:
confidence_interval

(1.0901973384384906, 1.3871359948948425)

The 99% confidence interval using the sample standard deviation is approximately (1.09million, 1.38 million) characters.

Steps taken:

1. Define the sample data.
2. Calculate sample mean and standard deviation.
3. Determine the degree of freedom(df).
4. Calculate the t-critical.
5. Calculate the margin of error.
6. Calculate the confidence interval.  

Rationale for using t-disteibution:



* The sample size is small (n=15).When the sample size is less than 30 we generally use t-distribution.  
*   When the population standard deviation is not know we use the sample standard deviation. so t-distribution is used.

   





**b**. **Build 99% Confidence Interval Using Known Population Standard Deviation**

If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.


In [14]:
# given data
s=0.2 # population standard deviation
# CI=0.99
# alpha=0.01
# n=15



In [15]:
z_critical=stats.norm.ppf(1-alpha/2) # calculates the critical value of the z-distribution for a two-tailed test

In [16]:
z_critical

2.5758293035489004

In [17]:
Margin_of_Error=z_critical*(s/np.sqrt(n)) # calculates the margin of error

In [18]:
pop_mean_lower_limit=x_bar-Margin_of_Error # calculates the lower limit
pop_mean_upper_limit=x_bar+Margin_of_Error # calculates the upper limit

In [19]:
print(pop_mean_lower_limit) # prints lower limit
print(pop_mean_upper_limit) # prints upper limit

1.1056514133957607
1.3716819199375725


In [20]:
confidence_interval=(pop_mean_lower_limit,pop_mean_upper_limit) # calculates the confidence interval

In [21]:
confidence_interval # prints confidence interval

(1.1056514133957607, 1.3716819199375725)

The 99% confidence interval for the mean number of characters printed before failure, using the known population standard deviation of
0.2 million characters, is approximately (1.10 million, 1.37 million) characters.